### PR TITLE
Return errors performing subsets as an output

### PR DESCRIPTION
**Pull request by jonescc**
Return errors running gogoduck/ncdfgenerator subsets as an output rather than a ProcessException.

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java [hunk 1]
```diff
@@ -1,47 +1,49 @@
 package au.org.emii.wps;
 
 import java.io.File;
+import java.nio.file.DirectoryStream.Filter;
+import java.util.HashMap;
```
Comment (dnahodil) [inline src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java]: These 2 imports are unused

File: src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
-import org.geoserver.wps.WPSException;
-import org.geoserver.wps.WPSInfo;
-import org.opengis.util.ProgressListener;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
+import org.geoserver.wps.gs.GeoServerProcess;
+import org.geoserver.wps.process.FileRawData;
+import org.geoserver.wps.process.RawData;
+import org.geoserver.wps.process.StringRawData;
```
Comment (dnahodil) [inline src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java]: This import is unused

File: src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java [hunk 3]
```diff
@@ -55,46 +57,54 @@
     public GoGoDuckProcess(WPSResourceManager resourceManager, GeoServerImpl geoServer, GeoServerResourceLoader resourceLoader) {
```
Comment (dnahodil) [inline src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java]: The `geoServer` parameter is now unused, can get remove it from the method signature?

File: src/extension/wps/src/main/java/au/org/emii/wps/SubsetExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.Map;
+
+import org.geoserver.wps.process.RawData;
+import org.geoserver.wps.process.StringRawData;
+import org.opengis.util.ProgressListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SubsetExceptionHandler {
+    private static final Logger logger = LoggerFactory.getLogger(SubsetOperation.class);
```
Comment (dnahodil) [inline src/extension/wps/src/main/java/au/org/emii/wps/SubsetExceptionHandler.java]: This logger is created for the wrong class

File: src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java [hunk 5]
```diff
@@ -1,47 +1,49 @@
...
-import javax.xml.xpath.XPathFactory;
+import au.org.emii.gogoduck.worker.GoGoDuck;
+import au.org.emii.gogoduck.worker.URLMangler;
 
 @DescribeProcess(title="GoGoDuck", description="Subset and download gridded collection as NetCDF files")
-public class GoGoDuckProcess implements GeoServerProcess {
+public class GoGoDuckProcess implements GeoServerProcess, SubsetOperation {
+    private static final String RESULT_TYPE = "application/x-netcdf";
+
     private static final Logger logger = LoggerFactory.getLogger(GoGoDuck.class);
```
Comment (dnahodil) [inline src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java]: I know you didn't write this but I see that this logger is also created for the wrong class.

Could we fix this one too, please.

**Comment by dnahodil** [comment]
Changes look good @jonescc. A few comments there, mostly pretty small stuff.

**Comment by jonescc** [comment]
@dnahodil - have made suggested changes.  Also changed the error output to be a literal output rather than a complex output and to ensure there is always an error message when an exception has occured

**Comment by dnahodil** [comment]
Looks good!

**Pull request merged**